tests/qdwh_test.py

```diff
@@ -129,14 +124,11 @@ def testQdwhWithUpperTriangularInputAllOnes(self, m, n, log_cond):
       self.assertAllClose(
           actual_results, expected_results, rtol=rtol, atol=1E-5)
 
-  @parameterized.named_parameters(jtu.cases_from_list(
-      {  # pylint:disable=g-complex-comprehension
-          'testcase_name': '_m={}_by_n={}_log_cond={}_padding={}'.format(
-              m, n, log_cond, padding),
-          'm': m, 'n': n, 'log_cond': log_cond, 'padding': padding}
-      for m, n in zip([6, 8], [6, 4])
-      for padding in (None, (3, 2))
-      for log_cond in np.linspace(1, 4, 4)))
+  @jtu.sample_product(
+    [dict(m=m, n=n) for m, n in zip([6, 8], [6, 4])],
```

Comment:
nit: `[{'m': 6, 'n': 6}, {'m': 8, 'n': 4}]` would be shorter and more direct

Reply:
Changed to avoid the `zip`, but I find it verbose to write out the names for each shape, especially if we ever add more test cases.